modules/wallet/update.go

```diff
@@ -344,66 +344,109 @@ func (w *Wallet) ProcessConsensusChange(cc modules.ConsensusChange) {
+			}
 		}
 
-		// only create a ProcessedTransaction if txn is relevant
-		if !relevant {
-			continue
-		}
+		// Set the unconfirmed preocessed transactions to the pruned set.
+		w.unconfirmedProcessedTransactions = newUPT
+	}
 
-		pt := modules.ProcessedTransaction{
-			Transaction:           txn,
-			TransactionID:         txn.ID(),
-			ConfirmationHeight:    types.BlockHeight(math.MaxUint64),
-			ConfirmationTimestamp: types.Timestamp(math.MaxUint64),
-		}
-		for _, sci := range txn.SiacoinInputs {
-			pt.Inputs = append(pt.Inputs, modules.ProcessedInput{
-				ParentID:       types.OutputID(sci.ParentID),
-				FundType:       types.SpecifierSiacoinInput,
-				WalletAddress:  w.isWalletAddress(sci.UnlockConditions.UnlockHash()),
-				RelatedAddress: sci.UnlockConditions.UnlockHash(),
-				Value:          spentSiacoinOutputs[sci.ParentID].Value,
-			})
-		}
-		for i, sco := range txn.SiacoinOutputs {
-			pt.Outputs = append(pt.Outputs, modules.ProcessedOutput{
-				ID:             types.OutputID(txn.SiacoinOutputID(uint64(i))),
-				FundType:       types.SpecifierSiacoinOutput,
-				MaturityHeight: types.BlockHeight(math.MaxUint64),
-				WalletAddress:  w.isWalletAddress(sco.UnlockHash),
-				RelatedAddress: sco.UnlockHash,
-				Value:          sco.Value,
-			})
+	// Scroll through all of the diffs and add any new transactions.
+	for _, uts := range diff.AppliedTransactions {
+		// Mark all of the transactions that appeared in this set.
+		//
+		// TODO: Technically only necessary to mark the ones that are relevant
+		// to the wallet, but overhead should be low.
+		w.unconfirmedSets[uts.ID] = uts.IDs
+
+		// Get the values for the spent outputs.
+		spentSiacoinOutputs := make(map[types.SiacoinOutputID]types.SiacoinOutput)
+		for _, scod := range uts.Change.SiacoinOutputDiffs {
+			// Only need to grab the reverted ones, because only reverted ones
+			// have the possibility of having been spent.
+			if scod.Direction == modules.DiffRevert {
+				spentSiacoinOutputs[scod.ID] = scod.SiacoinOutput
+			}
 		}
-		for _, fee := range txn.MinerFees {
-			pt.Outputs = append(pt.Outputs, modules.ProcessedOutput{
-				FundType: types.SpecifierMinerFee,
-				Value:    fee,
-			})
+
+		// Add each transaction to our set of unconfirmed transactions.
+		for i, txn := range uts.Transactions {
+			// determine whether transaction is relevant to the wallet
+			relevant := false
+			for _, sci := range txn.SiacoinInputs {
+				relevant = relevant || w.isWalletAddress(sci.UnlockConditions.UnlockHash())
+			}
+			for _, sco := range txn.SiacoinOutputs {
+				relevant = relevant || w.isWalletAddress(sco.UnlockHash)
+			}
+
+			// only create a ProcessedTransaction if txn is relevant
+			if !relevant {
+				continue
+			}
+
+			pt := modules.ProcessedTransaction{
+				Transaction:           txn,
+				TransactionID:         uts.IDs[i],
+				ConfirmationHeight:    types.BlockHeight(math.MaxUint64),
+				ConfirmationTimestamp: types.Timestamp(math.MaxUint64),
```

Comment:
I think `math.MaxUint64` is an untyped constant so an explicit cast isn't necessary. NBD